articles/connections/apple-siwa/troubleshooting.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Troubleshooting Sign in With Apple
+description: General troubleshooting steps for web and native apps using the Apple connection
+topics:
+  - troubleshooting
+  - errors
+  - authentication
+  - connections
+  - social
+  - apple
+contentType:
+  - how-to
+useCase: 
+  - troubleshooting
+---
+
+# Troubleshooting Sign in with Apple
+
+## Configuration elements
+
+Ensure that the correct configuration elements are in place, both in the Auth0 Management Dashboard and in the Apple Developer Settings Console. Common configuration problems include:
+
+- **Using the wrong identifier**: Remember that Apple App IDs (also known as App Bundle Identifiers) need to be [configured in Auth0's advanced application settings](/connections/apple-siwa/add-siwa-to-native-app).  Service IDs, which are used to configure web apps, need to be [configured in connection settings](/connections/apple-siwa/add-siwa-to-native-app).  Switching these identifiers will result in failures.
+- **Missing return URLS for Web Apps**: When using Sign In with Apple for web apps, the Auth0 callback endpoint must be added to the list of Return URLs in the Apple Developer Settings Console.  When not using custom domains, this will take the format: `https://TENANT.auth0.com/login/callback`.
+
+::: warning
+Remember that it's not possible to test native apps from the Auth0 Management Dashboard.  The **Try** button in the Apple connection settings only tests the web app flow.  This is due to the fact that real devices are required for interaction with the Apple IdP using an App ID.
+:::
+
+## Tenant logs
+
+If your application successfully initiated the login flow with Auth0, the results will be reflected in the tenant logs.  Native social exchanges will use the `sens` and `fens` event types to indicate success and failure (respectively), while web flows will use the standard `s` and `f` event types.  All tenant logs interacting with the Apple IdP will use the connection value of `apple`.
+
+## Types of errors
+
+The following errors may be returned from the Apple IdP.  Auth0 will relay both status codes and error messages from Apple should a request fail.
+
+| Error | Status Code | Description |
+| - | - | - |
+| `invalid_request` | 400 | The request parameters were incomplete or incorrect |
+| `invalid_grant` | 400 | The authorization code or refresh token presented to the Apple IdP is not valid |
```

Comment:
This makes it sound like we are sending the refresh token to Apple. I would probably remove the reference to refresh tokens completely and clarify that this is specific to the Native social login exchange

Reply:
I see what you're saying.  It *is* what Apple would send us in the event that happened, but TMI here could definitely cause confusion.  I'll take it out (on a subsequent PR).